base/task.jl

```diff
@@ -446,10 +446,11 @@ function enq_work(t::Task)
     (t.state == :runnable && t.queue === nothing) || error("schedule: Task not runnable")
     tid = (t.sticky ? Threads.threadid(t) : 0)
```

Comment:
I think a better version of this function would be:
```suggestion
if t.sticky
    tid = Threads.threadid(t)
    if tid == 0
        tid = Threads.threadid()
    end
    push!(Workqueues[tid], t)
else
    tid = 0
    ccall(:jl_enqueue_task, Cvoid, (Any,), t)
end
```